lib/target.py

```diff
@@ -0,0 +1,54 @@
+import socket
+import logging
+import datetime
+from netaddr import valid_ipv4
+from urllib.parse import urlparse
+
+
+class Target:
+
+    # Here, tasklist is a list of Task objects
+    def __init__(self, target, results_dict={}):
+        self.targetname = target
+        self.type = ""
+        self.tasklist = []
+        self.resultsdict = results_dict
+
+    def isValid(self):
+        # A target can be 2 things:
+        # 1. FQDN
+        # 2. IPv4 address
+
+        if not isinstance(self.targetname, str):
+            return False
+
+        starts_with_anti_patterns = ["127.0.0", "10.", "172.", "192.168", "169.254.169.254"]
```

Comment:
Should we add common schemes to the anti_patterns list, like "http", "ftp", "https", etc.?  I  feel like it takes less pressure off the network I/O dependant checks and will be a faster fail.

Reply:
This is a really good idea actually, will implement. Thanks!